esm_runscripts/event_handlers.py

```diff
@@ -0,0 +1,36 @@
+import signal
+import sys
+import os
+
+def signal_listener():
+    """Listens to the signals (eg. CTRL-C press, ...) and calls the handler
+    for that signal"""
+    
+    # 1) handler for the SIGINT (CTRL-C)
+    signal.signal(signal.SIGINT, handle_sigint)
+    
+    # other signal handlers may go here, eg. SIGHUP, SIGSTOP, ...)
+    # each signal is handled by a function handle_<signal>, eg. handle_sigint()
+
+
+def handle_sigint(signum, stack_frame):
+    """Handles the SIGINT signal (aka CTRL-C press on UNIX/LINUX)
+    
+    Parameters
+    ----------
+    signum : int
+        signal number.
+    stack_frame : Python frame object
+        current stack frame
+    
+    More info: https://docs.python.org/3/library/signal.html
```

Comment:
Stick that under a Notes heading:
```
Notes
------
The signal library is documented here: https://docs.python.org/3/library/signal.html
```

Reply:
Here is the file call hierarchy: 

1. first you call the executable (esm_runscripts, eg. ~/.local/bin/esm_runscripts).
2. This calls `load_entry_point('esm-runscripts', 'console_scripts', 'esm_runscripts')()`
3. This calls `'console_scripts': ['esm_runscripts=esm_runscripts.cli:main', ],`
4. Enter `cli.py` that I modified.

Obviously, it takes some time until the execution reaches `cli.py`. But what happens if someone executes `esm_runscripts` can then presses `CTRL-C` within few seconds. It will print the nasty error stack since the event listener has not started yet. 

On the other hand, if I place the `event_handler` to the top of the `esm_runscripts` CLI executable (that is automatically created by the Python setuptools module then there will be a greater chance to catch that early `CTRL-C`.

According to the setuptools documentation, it will generate an automatic CLI executable if you use the `console_scripts` entry point. You can provide your own executable script if you use `scripts` instead.

Here is a nice read: https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

This is a personal preference of course. Sometimes I hit `CTRL-C` within 1-2 seconds since I realize that I forgot something